pkg/controller/external.go

```diff
@@ -155,13 +159,18 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
 
+	policySet := sets.New[xpv1.ManagementAction](tr.GetManagementPolicies()...)
+
 	// Note(turkenh): We don't need to check if the management policies are
 	// enabled or not because the crossplane-runtime's managed reconciler already
-	// does that for us. In other words, if the management policy is set to
-	// "ObserveOnly" without management policies being enabled, the managed
+	// does that for us. In other words, if the management policies are set
+	// without management policies being enabled, the managed
 	// reconciler will error out before reaching this point.
 	// https://github.com/crossplane/crossplane-runtime/pull/384/files#diff-97300a2543f95f5a2ada3560bf47dd7334e237e27976574d15d1cddef2e66c01R696
-	if tr.GetManagementPolicy() == xpv1.ManagementObserveOnly {
+	// Note (lsviben) We are only using import instead of refresh if the
+	// management policies do not contain create or update as they need the
+	// required fields to be set, which is not the case for import.
+	if !policySet.HasAny(xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate, xpv1.ManagementActionAll) {
```

Comment:
Isn't this going to result in an import if, for example, the management policy is set only to `Delete`?  Do we rely on the `managed.defaultSupportedManagementPolicies` here (looks like the only supported default policies not containing these are the `Observe` and the empty policy)? I assume with the empty policy we don't expect to reach here and for the remaining `Observe`, import is the right thing to do.

But what will happen, if for instance, the provider configures the supported management policies to include some other set? Do we rely specifically on the default supported policies here?

Reply:
As in crossplane-runtime, we decided not to support & test initially all possible combinations. It's true that Provider developers can create their own set of supported management polices, but if something is not working for them out of the box, I would expect them to create a ticket to support some combination or contribute it.  

The options which are left when we remove Create and Update are (and not currently supported by default):
```
["Observe", "LateInitialize", "Delete"]
["Observe", "LateInitialize"]
["Observe", "Delete"]
["Delete", "LateInitialize"]
["Delete"]
["LateInitialize"]
```

Supported by default:
```
["Observe"]
[""]
```
From my perspective any without "Observe" does not make a lot of sense (or I dont see the use case) so what is left is :
```
["Observe", "LateInitialize", "Delete"]
["Observe", "LateInitialize"]
["Observe", "Delete"]
```

I dont see why `["Observe", "Delete"]` should not work. Only thing I see is that if we add `LateInitialize` that the late initialization wont happen. I can add the code for late initialization to Import if that is what we want. Not sure if we want to support this right away though.

Reply:
I think what we have implemented here is correct, given that what's left for here is the `{Observe}` and the empty set, `{}` (in which case we will just pause reconciliation I assume). What worries me is that we have an API to override the set of supported policies for an upjet-based provider, and this implementation makes assumptions on the set of supported policies. In principle, I think it would be safer to disallow (at least for now until we have a good use case) the API that allows changing the supported policies but I don't certainly think that it's a blocker for this PR.